ccdproc/bitmask.py

```diff
@@ -0,0 +1,473 @@
+
+            >>> bm.to_mask(bitpos=[0, 1])
+            array([False,  True,  True,  True, False,  True,  True,  True], dtype=bool)
+        """
+        bitmask = self.data
+        # Make sure one and only one of the three options to provide the
+        # "active" flags is actually given.
+        if sum(inp is not None for inp in (bitpos, bitval, bitname)) != 1:
+            raise TypeError('One and only one of "bitpos", "bitval" or '
+                            '"bitname" may be not "None".')
+
+        # So that the logic and validation is only applied once process
+        # first "bitname" to "bitpos" and then "bitpos" to "bitval". So each
+        # step is quite short and well defined and the logic can be implemented
+        # only in the "bitval" branch.
+
+        # Map the names to bit positions
+        if bitname is not None:
+            try:
+                bitpos = [self.name_map[name] for name in bitname]
+            except KeyError:
+                raise ValueError(
+                    'The "name(s)" {} were not found in "self.name_map" ({})'
+                    .format([name for name in bitname
+                             if name not in self.name_map],
+                            self.name_map))
+            bitname = None
+
+        # Map the bit positions to bit values
+        if bitpos is not None:
+            try:
+                bitval = [_POWERS[pos] for pos in bitpos]
+            except KeyError:
+                raise ValueError(
+                    'The "bit position(s)" {} are not in the allowed range of '
+                    'values.'.format([pos for pos in bitpos
+                                      if pos not in _POWERS]))
+            bitpos = None
+
+        # If bitname or bitvals were given these are now converted to bitvalues
+        # so we can do all the real processing in here.
+        if bitval is not None:
+            # There is no zero'th flag so subtract one from the representable
+            # flags to get the real number of flags
+            maximum = _REPRESENTABLE_FLAGS[bitmask.dtype.name] - 1
+            # Make sure the maximum value doesn't exceed the limit given by the
+            # dtype of the bitmask
+            if max(bitval) > _POWERS[maximum]:
+                raise ValueError(
+                    'The range of values ({}) exceeds the limit of the data '
+                    'type of the bitmask ({}).'
+                    .format(max(bitval), _POWERS[maximum]))
+            uniques = set(bitval)
+            # Make sure only valid powers of two are inside the bitvals
+            if not uniques.issubset(_POWERS):
+                raise ValueError(
+                    'The "bitvals" contain illegal (not powers of two) values:'
+                    ' {}'.format(sorted(uniques.difference(_POWERS))))
+
+            # Independant of the flag interpretation we need to sum all
+            # unique bitvalues
+            cmpbitval = sum(uniques)
```

Comment:
I think having users to provide ``cmpbitval`` as an input argument is much more useful then, e.g., ``bitname`` (in fact, I can't even imagine what who and when would like to use ``bitname``).